test/error-messages/nested-invalid-value.txt

```diff
@@ -0,0 +1,9 @@
+Invalid value
+
+concat :: Array a -> Array a -> Array a
+                     ^^^^^^^
+                        1
+
+1) "ab" :: String
+
+The value at position 1 is not a member of (Array Number).
```

Comment:
This message does not explain how we determined `a` to be `Number`. Ideally I'd like to see something like this:

    Invalid value

    concat :: Array a -> Array a -> Array a
              ^^^^^^^    ^^^^^^^
                 1          2

    1)  [1, 2] :: Array Number
    2)  "ab" :: String

    The value at position 2 is not a member of (Array Number).

I realize this may be tricky. It's very much a nice-to-have rather than a must-have.